compiler/modules/front.syntax/ast.bal

```diff
@@ -524,9 +518,22 @@ public type BinaryTypeDesc record {|
     TypeDesc right;
 |};
 
-public type UnaryTypeDesc record {|
+public type UnaryTypeDesc PrefixTypeDesc|PostfixTypeDesc;
+
+public type PrefixTypeOp "!"|"(";
+
+public type PrefixTypeDesc record {|
     *PositionFields;
-    UnaryTypeOp op;
+    PrefixTypeOp op;
+    TypeDesc td;
+|};
+
+public const PostfixTypeOp = "?";
+
+public type PostfixTypeDesc record {|
```

Comment:
Remember that type names are just aliases.  So what you have defined is completely equivalent to having a UnaryTypeDesc record with an `op` field of `"!"|"?"|"("`. That is OK as a design but you can be written much more simply than what you have here.

Reply:
fixed